### PR TITLE
nan consistency in lst-binning

### PR DESCRIPTION
**Pull request by matyasmolnar**
Quick fix to issue #744 

[edit by nk]:
to fix this we use nansum to ignore nans in the weighted mean, which preserves nan for downstream nanstd estimator

### CODE REVIEW

**Comment by nkern** [comment]
Great catch! thanks @matyasmolnar

**Review by nkern — APPROVED** [review]

**Comment by matyasmolnar** [comment]
@nkern I've just added a couple of lines to filter out (the many) warnings raised when an all nan row is encountered in the `np.nanstd` statistics [here](https://github.com/HERA-Team/hera_cal/blob/7829a309519cef386d0053c33673b78e80901dde/hera_cal/lstbin.py#L380)

**Comment by nkern** [comment]
@matyasmolnar looks great, but can we keep the nanned data as 1.0? there are good reasons to keep fake data as 1 not zero, mainly in case we want to divide two datasets and not incur divide by zero errors. after that reversion I think this is good to go!

File: hera_cal/utils.py
Comment (nkern) [inline hera_cal/utils.py:1321]: @jsdillon @aewallwi we are proposing to change the behavior of `red_average` such that pixels that are fully flagged are set to 1.0 instead of 0.0. this is more consistent with what the LST binner does. can you comment on any possible ramifications of this?

In theory, any downstream block that honors flags should not be affected. the reason to move to 1.0 instead of 0.0 is that it is 1. consistent with lst binner and 2. plays nicer with division

File: hera_cal/tests/test_utils.py
Comment (nkern) [inline hera_cal/tests/test_utils.py:614]: the consequence of changing that line in `utils.red_average` is a need to update this unit test, which seems fairly isolated to this specific functionality and wouldn't affect other parts of hera_cal.

**Comment by jsdillon** [comment]
I don't feel too strongly about this, but the convention in `redcal` is that flagged visibilities are 0 and flagged gains are 1. 

That said, I don't think there will be any downstream consequences...

File: hera_cal/utils.py
Comment (aewallwi) [reply hera_cal/utils.py:1321]: I think this should be fine. I approve.

**Comment by jsdillon** [comment]
Are we waiting on anyone's approval here?

**Comment by matyasmolnar** [comment]
Just waiting on approval from @aewallwi I believe

**Review by aewallwi — APPROVED** [review]

**Pull request merged**
